fdt.hh

```diff
@@ -616,6 +621,16 @@ class node
 	{
 		children.push_back(std::move(n));
 	}
+	/**
+	 * Deletes a child from this node.
+	 */
+	inline void delete_child(node_ptr &n)
```

Comment:
This is used in only one place, so you should probably just inline it.

Reply:
The problem I had there was that `children` is private to the node, so accessing it from the device tree for the purposes of removing elements was a no-fly since I didn't want to change access rights or 'friend' it up.